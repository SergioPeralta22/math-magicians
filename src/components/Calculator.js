import { useState } from 'react';
import buttons from '../logic/buttons';
import calculate from '../logic/calculate';
import Button from './Button';

function Calculator() {
  const [calculation, setCalculation] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleClick = (e) => {
    setCalculation(calculate(calculation, e.target.textContent));
  };

  const { total, next } = calculation;
  const display = next || total || '0';

  return (
    <div className="calculator">
      <input
        type="text"
        className="calculator-display"
        value={display}
        readOnly
      />
      <div className="calculator-buttons">
        {buttons.map((button) => (
          <Button
            key={button.name}
            className={button.className}
            onClick={handleClick}
            name={button.name}
          />
        ))}
      </div>
    </div>
  );
}

export default Calculator;
