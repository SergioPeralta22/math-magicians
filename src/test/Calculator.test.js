import renderer from 'react-test-renderer';
import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Calculator from '../components/Calculator';

describe('test snapshop', () => {
  it('should render Calculator', () => {
    const renderCalculator = renderer.create(<Calculator />).toJSON();
    expect(renderCalculator).toMatchSnapshot();
  });
});

describe('test with testing library', () => {
  test('should work for addition', () => {
    render(<Calculator />);
    fireEvent.click(screen.getByText('8'));
    fireEvent.click(screen.getByText('+'));
    fireEvent.click(screen.getByText('8'));
    fireEvent.click(screen.getByText('='));
    expect(screen.getByRole('textbox')).toHaveValue('16');
  });
});
