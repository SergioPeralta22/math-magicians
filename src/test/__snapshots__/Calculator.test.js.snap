// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshop should render Calculator 1`] = `
<div
  className="calculator"
>
  <input
    className="calculator-display"
    readOnly={true}
    type="text"
    value="0"
  />
  <div
    className="calculator-buttons"
  >
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="btn0"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
